arangod/Transaction/Methods.cpp

```diff
@@ -2343,19 +2343,27 @@ Future<Result> Methods::replicateOperations(
     // We drop all followers that were not successful:
     for (size_t i = 0; i < followerList->size(); ++i) {
       auto const& tryRes = responses[i];
-      network::Response const& resp = tryRes.get();
-
       bool replicationWorked = false;
-      if (resp.error == fuerte::Error::NoError) {
-        replicationWorked = resp.response->statusCode() == fuerte::StatusAccepted ||
-                            resp.response->statusCode() == fuerte::StatusCreated ||
-                            resp.response->statusCode() == fuerte::StatusOK;
-        if (replicationWorked) {
-          bool found;
-          resp.response->header.metaByKey(StaticStrings::ErrorCodes, found);
-          replicationWorked = !found;
+      try {
+        // Note that it is possible that the future was resolved with an exception
+        // which will be thrown when we ask for the response, we need to catch
+        // the exception here.
```

Comment:
An alternative approach would be to explicitly check `if (tryRes.hasValue()) { /* handle response */ } else { /* log and fall through */ }`.

Reply:
That is right. I do not actually know which I prefer. In any case, I think we had decided against the use of exceptions for error handling a long time ago...